tests/e2e/repo_finder/repo_finder.py

```diff
@@ -62,6 +62,16 @@ def test_repo_finder() -> int:
     if not find_repo(PackageURL.from_string("pkg:cargo/rand_core")):
         return os.EX_UNAVAILABLE
 
+    # Test priority ordering of Repo Finder using artifact with differing URLs in the POM's SCM section.
```

Comment:
Thanks for adding the test. What we are trying to test here is good. However, I think it relies a lot on the outside states:
- The pom file stored in maven central for `pkg:maven/org.glassfish.jersey/project@3.0.12` which we obtain and inspect.
- The network connection.
Because the behavior we are trying to test doesn't have to rely on those things (we are testing if given a pom file, the order of `repo_pom_paths` is reflected correctly in the order of valid scm urls extracted). I believe it would be even better if we make this a unit test with a pom file (you can reuse from `pkg:maven/org.glassfish.jersey/project@3.0.12` or crafted a custom pom.xml file) so that the test is deterministic (or you can add an additional unit test and keep this e2e test). 
Please let me know what you think about it.

Reply:
It does rely on a very specific POM to properly test the ordering. While that POM is unlikely to change, it would be troublesome to find another in the case that it did. Therefore I think you are right that this test can be replaced with a unit test and custom POM.